sql/tdigest.sql

```diff
@@ -0,0 +1,204 @@
+  finalfunc = _timescaledb_internal.tdigest_final,
+  parallel = safe
+);
+
+CREATE AGGREGATE tdigest_create(float8, integer) (
+  sfunc = _timescaledb_internal.tdigest_trans,
+  stype = internal,
+  serialfunc = _timescaledb_internal.tdigest_serialize,
+  deserialfunc = _timescaledb_internal.tdigest_deserialize,
+  combinefunc = _timescaledb_internal.tdigest_combine,
+  finalfunc = _timescaledb_internal.tdigest_final,
+  parallel = safe
+);
+
+-- Comparison Operators
+
+CREATE FUNCTION _timescaledb_internal.tdigest_cmp(tdigest, tdigest)
+RETURNS integer
+AS '@MODULE_PATHNAME@', 'ts_tdigest_cmp_sql'
+LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_equal(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_equal_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_equal_exact(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_equal_exact_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_lt(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_lt_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_gt(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_gt_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_ge(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_ge_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION _timescaledb_internal.tdigest_le(@extschema@.tdigest, @extschema@.tdigest)
+RETURNS boolean
+AS '@MODULE_PATHNAME@', 'ts_tdigest_le_sql'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE OPERATOR < (
+    LEFTARG = @extschema@.tdigest,
+    RIGHTARG = @extschema@.tdigest,
+    PROCEDURE = _timescaledb_internal.tdigest_lt,
+    commutator = >,
+    NEGATOR = >=
+);
+
+CREATE OPERATOR <= (
+    leftarg = @extschema@.tdigest,
+    rightarg = @extschema@.tdigest,
+    procedure = _timescaledb_internal.tdigest_le,
+    commutator = >=,
+    NEGATOR = >
+);
+
+CREATE OPERATOR = (
+    leftarg = @extschema@.tdigest,
+    rightarg = @extschema@.tdigest,
+    procedure = _timescaledb_internal.tdigest_equal,
+    commutator = =,
+    negator = !=
+);
+
+CREATE OPERATOR >= (
+    LEFTARG = @extschema@.tdigest,
+    RIGHTARG = @extschema@.tdigest,
+    PROCEDURE = _timescaledb_internal.tdigest_ge,
+    commutator = <=,
+    NEGATOR = < 
+);
+
+CREATE OPERATOR > (
+    LEFTARG = @extschema@.tdigest,
+    RIGHTARG = @extschema@.tdigest,
+    PROCEDURE = _timescaledb_internal.tdigest_gt,
+    commutator = <,
+    NEGATOR = <= 
+);
+
+--- Implement complete operator class to support use of DISTINCT and ORDER BY
+--- Metric for total order is total number of data points added to a TDigest
+CREATE OPERATOR CLASS @extschema@.tdigest_ops
```

Comment:
do we have a test for this?

Reply:
is it possible to create a table with "tdigest" type as column and an index on it. That will probably exercise this code.

Reply:
Done using ORDER BY in LIMIT statements in both T-Digest test and cont_agg_dump test